apps/wpcom-block-editor/src/wpcom/features/tracking.js

```diff
@@ -16,6 +16,25 @@ import delegateEventTracking from './tracking/delegate-event-tracking';
 // Debugger.
 const debug = debugFactory( 'wpcom-block-editor:tracking' );
 
+/**
+ * Global handler.
+ * Use this function when you need to inspect the block
+ * to get specific data and populate the record.
+ *
+ * @param {object} block - Block object data.
+ * @param {object} parentBlock - Block object data.
+ * @returns {object} Record properties object.
+ */
+function globalEventPropsHandler( block, parentBlock ) {
+	// Pick up block variation slug from embed block.
+	if ( ! block?.name || block.name !== 'core/embed' || ! block?.attributes.providerNameSlug ) {
```

Comment:
Do you see any way to track the `variation_slug` for any block that uses variations, rather than having to opt in for each individual block?

Reply:
We could check all blocks, but I think the attribute name is arbitrary, meaning that maybe the attribute name of other block variations won't match with `variation_slug`. I'm discarding other blocks to make this process as fast as possible.

Reply:
I've researched a little bit about the block API and it doesn't seem to provide a way to get the variation of the block, in the edition context. The block variation lives in the block inserter context.

Reply:
In that case, how would you modify this to make it easy to opt in other blocks for tracking their variations?

Reply:
I'm afraid we'll have to deal with each block type separately. Let me take a look at social icons, for instance.

Reply:
Added social-link variation slug a4ddcf7

Reply:
That makes sense. At some point it'll make sense to have a way to "register" a block name and the appropriate attribute for tracking the variation, but this works for a first pass. If you get D50097-code updated I'll review it.